11-payment-encoding.md

```diff
@@ -0,0 +1,339 @@
+* `20m`: amount (20 milli-bitcoin)
+* `1`: Bech32 separator
+* `q`: version (0)
+* `pvj6chq`: timestamp (1496146656)
+* `qqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqypq`: payment hash.
+* `h`: tagged field: hash of description
+  * `p5`: field length (`p` = 1, `5` = 20. 1 * 32 + 20 == 52)
+  * `8yjmdan79s6qqdhdzgynm4zwqd5d7xmw5fk98klysy043l2ahrqs`: SHA256 of 'One piece of chocolate cake, one icecream cone, one pickle, one slice of swiss cheese, one slice of salami, one lollypop, one piece of cherry pie, one sausage, one cupcake, and one slice of watermelon'
+* `cpt2ld45dqewrllnmf6hj355nfeypurkr6a2d0neyq2e6g9u6ur9tl7e7drhglfrn9yxk2cdujutuqksx2agqv8mphl0mzjrwm6k59qq`: signature
+* `2mnedn`: Bech32 checksum
+
+> ### The same, on testnet, with a fallback address mk2QpYatsKicvFVuTAQLBryyccRXMUaGHP
+> lntb20m1qpvj6chqqqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqypqfpp3x9et2e20v6pu37c5d9vax37wxq72un98hp58yjmdan79s6qqdhdzgynm4zwqd5d7xmw5fk98klysy043l2ahrqsqsj75rjv443nrh8gu5xutlyyqyx6ul76m2rx87yxr5gdfagyywn3s6wtpfrl6elncce7rmh6kndvr5nur76w9u7z0k3gq93fyfpu9zqq3jp9as
+
+Breakdown:
+
+* `lntb`: prefix, lightning on bitcoin testnet
+* `20m`: amount (20 milli-bitcoin)
+* `1`: Bech32 separator
+* `q`: version (0)
+* `pvj6chq`: timestamp (1496146656)
+* `f`: tagged field: fallback address
+  * `pp`: field length (`p` = 1. 1 * 32 + 1 == 33)
+  * `3x9et2e20v6pu37c5d9vax37wxq72un98`: `3` = 17, so P2PKH address
+* `h`: tagged field: hash of description...
+* `qsj75rjv443nrh8gu5xutlyyqyx6ul76m2rx87yxr5gdfagyywn3s6wtpfrl6elncce7rmh6kndvr5nur76w9u7z0k3gq93fyfpu9zqq`: signature
+* `3jp9as`: Bech32 checksum
+
+> ### On mainnet, with fallback address 1RustyRX2oai4EYYDpQGWvEL62BBGqN9T with extra routing info to get to node 029e03a901b85534ff1e92c43c74431f7ce72046060fcf7a95c37e148f78c77255
+> lnbc20m1qpvj6chqqqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqypqrz0q20q82gphp2nflc7jtzrcazrra7wwgzxqc8u7754cdlpfrmccae92qgzqvzq2ps8pqqqqqq5qqqqqqcfpp3qjmp7lwpagxun9pygexvgpjdc4jdj85fhp58yjmdan79s6qqdhdzgynm4zwqd5d7xmw5fk98klysy043l2ahrqst7xnzx2sl9rkrc5lgzkcdqs57sj0s0vz8z9g2wk4hucfdrtvupkx93wqjcj8lpejzc95k4p4hw0qrfay5x36def3ret6yd9s0vqwtysqm5z5dw
+
+Breakdown:
+
+* `lnbc`: prefix, lightning on bitcoin mainnet
+* `20m`: amount (20 milli-bitcoin)
+* `1`: Bech32 separator
+* `q`: version (0)
+* `pvj6chq`: timestamp (1496146656)
+* `qqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqypq`: payment hash.
+* `r`: tagged field: route information
+  * `z0`: field length (`z` = 2, `0` = 15.  2 * 32 + 15 = 79)
+    `q20q82gphp2nflc7jtzrcazrra7wwgzxqc8u7754cdlpfrmccae92qgzqvzq2ps8pqqqqqq5qqqqqqc`: pubkey `029e03a901b85534ff1e92c43c74431f7ce72046060fcf7a95c37e148f78c77255`, `channel_id` 0102030405060708, `fee` 20 millisatoshi, `cltv_expiry_delta` 3.
+* `f`: tagged field: fallback address...
+* `h`: tagged field: hash of description...
+* `t7xnzx2sl9rkrc5lgzkcdqs57sj0s0vz8z9g2wk4hucfdrtvupkx93wqjcj8lpejzc95k4p4hw0qrfay5x36def3ret6yd9s0vqwtysq`: signature
+* `m5z5dw`: Bech32 checksum
+
```

Comment:
Likely related to the inconsistency mentioned above, the "r" field here is encoded with 32-bit (padded to 35-bit) fee and cltv which does not match the definition under Tagged Fields. And unless my manual bech32 decoding is wrong, the cltv in the example reads 24 rather than 3?